### PR TITLE
test: Add integration tests for node

### PR DESCRIPTION
**Pull request by mattcole19**
Integration tests against prism mock server for generated node code.

[Jira](https://issues.corp.twilio.com/browse/DI-1411)

### CODE REVIEW

File: examples/node/spec/integration/TestRequestClient.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import RequestClient from '../../lib/base/RequestClient';
+
+export default class TestRequestClient extends RequestClient {
+    constructor() {
+        super();
+    }
+
+    request(opts: object) {
+        opts.uri = opts.uri.replace("https", "http");
```
Comment (childish-sambino) [inline examples/node/spec/integration/TestRequestClient.ts:9]: I say we just update the `rest/*Base.ts` files to be `http` instead and then this isn't needed.

File: examples/node/lib/rest/Twilio.ts [hunk 2]
```diff
@@ -12,14 +17,14 @@ class Twilio {
...
 
-  request(opts: RequestOpts) {
-    return new RequestClient().request({
-      method: opts.method,
-      uri: opts.uri,
-      params: opts.params,
-      data: opts.data,
-    });
-  }
+  // request(opts: RequestOpts) {
```
Comment (childish-sambino) [inline examples/node/lib/rest/Twilio.ts:20]: Should uncomment/roll this back.

File: examples/node/spec/unit/credential.spec.ts [hunk 3]
```diff
@@ -3,7 +3,9 @@ import nock from "nock";
 import Twilio from "../../lib/rest/Twilio";
 
 describe("credential", () => {
-  const twilio = new Twilio();
+  const accountSid:string = 'AC12345678123456781234567812345678';
+  const authToken:string = 'CR12345678123456781234567812345678';
+  const twilio = new Twilio(accountSid, authToken);
```
Comment (childish-sambino) [inline examples/node/spec/unit/credential.spec.ts:8]: `username` and `password` are optional so can revert these.

File: examples/node/lib/rest/Twilio.ts (hunk 2 above)
Comment (JenniferMah) [inline examples/node/lib/rest/Twilio.ts:20]: Why does this need to be commented out?

Comment (mattcole19) [reply examples/node/lib/rest/Twilio.ts:20]: I tired using this initially but kept getting a "Invalid security scheme used" issue. Even when I added in the username and password options to the request

File: examples/node/spec/unit/credential.spec.ts (hunk 3 above)
Comment (mattcole19) [reply examples/node/spec/unit/credential.spec.ts:8]: extending BaseTwilio makes it so the username and password were required

File: examples/node/lib/rest/Twilio.ts (hunk 2 above)
Comment (childish-sambino) [reply examples/node/lib/rest/Twilio.ts:20]: Either way, should drop the commented-out code.

File: examples/node/spec/integration/TestRequestClient.ts (hunk 1 above)
Comment (childish-sambino) [reply examples/node/spec/integration/TestRequestClient.ts:9]: Can delete this now and just use `RequestClient`.

**Review by childish-sambino — APPROVED** [review]

**Comment by mattcole19** [comment]
These tests keep failing and I have an idea why, but I'm not sure how to fix it.

On my local machine, if I run the node-test service in docker-compose, the integration tests fail with a "Nock: no match for request" message. When I look at the prism server, none of the requests came through.

HOWEVER, if I stop and re-run the prism server, then run the node tests again, everything works fine (prism server gets hit).

Could this be an issue with prism not being ready in time...?

**Comment by childish-sambino** [comment]
@mattcole19 Yes, entirely possible. Workarounds in the past:
* [`77da272` (#269)](https://github.com/twilio/twilio-oai-generator/pull/269/commits/77da27288c7a87f05555a311ae212b9a66e11737)
* https://github.com/sendgrid/sendgrid-oai/blob/c3aaa432b769faa47285166aca17c7ed2ea71787/prism/docker-compose.yml#L8

**Comment by mattcole19** [comment]
Still unable to get these integration tests passing via github actions. They work locally, but only when I restart the prism server and re-run the node tests. Adding a "sleep 60" didn't resolve the issue :/

I'm really not sure how to proceed here...

**Comment by childish-sambino** [comment]
@mattcole19 Tests fail for me locally. Are you running `make test-docker`?

**Comment by mattcole19** [comment]
> @mattcole19 Tests fail for me locally. Are you running `make test-docker`?

Yes. They fail for me initially too, but when I restart the prism server and re-run the node service (typically via Docker desktop), they pass. Not sure why this "fixes" the issue

**Comment by childish-sambino** [comment]
I'm seeing nock issues. Looks like it's not being cleaned. Recommend sticking this at the top of all the files that use nock:
```js
afterAll(() => {
  nock.cleanAll();
});
```

**Review by childish-sambino — APPROVED** [review]

**Pull request merged**